lib/s3/binary_stream.ex

```diff
@@ -0,0 +1,18 @@
+defmodule S3.BinaryStream do
+  @moduledoc """
+  Provide an S3 object as a stream of binary chunks.
+  """
+
+  @doc """
+  Returns an S3 object as a stream of binary chunks.
+  """
+  @spec binary_chunks(String.t(), String.t(), ExAws.S3.download_file_opts()) :: Enumerable.t()
+  def binary_chunks(bucket, file, opts \\ []) do
+    download_file_fn = Keyword.get(opts, :download_file_fn, &ExAws.S3.download_file/4)
+    stream_fn = Keyword.get(opts, :stream_fn, &ExAws.stream!/1)
+
+    bucket
+    |> download_file_fn.(file, :memory, opts)
+    |> stream_fn.()
```

Comment:
I agree with you: given how simple this function is I don't think it's worth the effort to mock out the functions.